File: src/main/java/com/github/njuro/jard/config/security/methods/AuthorityAttribute.java
package com.github.njuro.jard.config.security.methods;

import com.github.njuro.jard.user.UserAuthority;
import java.io.Serial;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.security.access.ConfigAttribute;

/** Configuration attribute based on {@link UserAuthority}. */
@AllArgsConstructor
public class AuthorityAttribute implements ConfigAttribute {

  @Serial private static final long serialVersionUID = -8715812535674792037L;

  @Getter private final UserAuthority authority;

  @Override
  public String getAttribute() {
    return authority.name();
  }
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/config/security/methods/AuthorityAttribute.kt
package com.github.njuro.jard.rewrite.config.security.methods

import com.github.njuro.jard.user.UserAuthority
import org.springframework.security.access.ConfigAttribute

/** Configuration attribute based on [UserAuthority].  */
class AuthorityAttribute(val authority: UserAuthority) : ConfigAttribute {
    override fun getAttribute() = authority.name
}
